app/celer-sim/Transporter.cc

```diff
@@ -34,6 +34,22 @@ namespace celeritas
 {
 namespace app
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Print progress after N events when requested.
+ */
+void progress(EventId const id, size_type const num_primaries)
+{
+    CELER_EXPECT(num_primaries > 0);
+    CELER_LOG_LOCAL(status)
+        << "Event " + std::to_string(id.unchecked_get()) + ": transporting "
+               + std::to_string(num_primaries)
+               + (num_primaries == 1 ? " primary" : " primaries");
```

Comment:
If you use the stream insertion operator instead of concatenating strings with `+` you don't need the `to_string()` calls and avoid creating temporary strings:
```suggestion
        << "Event " << id.unchecked_get() << ": transporting " << num_primaries
        << (num_primaries == 1 ? " primary" : " primaries");
```